src/app/problems/problem-view/mcq-view-snippet/mcq-view-snippet.component.ts

```diff
@@ -43,7 +43,8 @@ export class McqViewSnippetComponent implements OnInit {
         if (!this.uqj.is_checkbox) {
             this.formData = this.formBuilder.group({
                 question: new FormControl(this.uqj.question.id),
-                solution: new FormControl(null, [Validators.required])
+                solution: new FormControl(null, [Validators.required]),
+                time_spent: parseInt(localStorage.getItem(this.uqj.id.toString()))
```

Comment:
This would get the time from storage when the question is loaded? Should this retrieval not be done until the question is submitted? It's also only being added for MCQ questions and not for checkbox questions.

Reply:
'time_spent' is retrieved again in OnSubmit(), I didn't remove the parameters of OnSubmit() cuz I'm not sure if that'll break things..